clients/roscpp/src/libros/service_server_link.cpp

```diff
@@ -362,14 +368,35 @@ bool ServiceServerLink::call(const SerializedMessage& req, SerializedMessage& re
     processNextCall();
   }
 
+  bool interrupted = false;
   {
     boost::mutex::scoped_lock lock(info->finished_mutex_);
 
+    using namespace boost::chrono;
+    steady_clock::time_point now = steady_clock::now();
+
     while (!info->finished_)
     {
-      info->finished_condition_.wait(lock);
+      if (timeout > 0)
+      {
+        steady_clock::time_point wall_time = now + milliseconds(static_cast<int>(timeout * 1000));
+
+        if (info->finished_condition_.wait_until(lock, wall_time) == boost::cv_status::timeout)
+        {
+          ROSCPP_LOG_DEBUG("Service [%s] call failed: no response for %fsec", service_name_.c_str(), timeout);
```

Comment:
@dirk-thomas I reduce this from `ROS_ERROR` to `ROSCPP_LOG_DEBUG`

Reply:
IMHO this should be a warning...

Reply:
Changed to `ROS_WARN`.

@dirk-thomas Do you agree with this?